packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -712,14 +747,43 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       return prevState;
     }
 
+    let maintainVisibleContentPositionAdjustment =
+      prevState.maintainVisibleContentPositionAdjustment;
+    const prevFirstItemKey = prevState.firstItemKey;
+    const newFirstItemKey = VirtualizedList._getItemKey(newProps, 0);
+    if (
+      newProps.maintainVisibleContentPosition != null &&
+      maintainVisibleContentPositionAdjustment == null &&
+      prevFirstItemKey != null &&
+      newFirstItemKey != null
+    ) {
+      // Fast path if items were added at the start of the list.
+      const hint = itemCount - prevState.renderMask.numCells();
```

Comment:
We won't do the lookup when items are added at the end of the list since the first item did not change (see https://github.com/facebook/react-native/pull/35993/files#diff-b450c5c4ec05ca5221b18a829f764b8397eed35a93f388893debb184591b0207R763). Maybe I should use if .. else block instead of ternary to make this logic more obvious?